src/devices/Mcp25xxx/Mcp25xxx.cs

```diff
@@ -0,0 +1,407 @@
+            {
+                VerifyPinConfigured(_reset, nameof(ResetPin));
+                _gpioController.Write(_reset, value);
+            }
+        }
+
+        /// <summary>
+        /// Reads the current value of Interrupt pin.
+        /// </summary>
+        public PinValue InterruptPin
+        {
+            get
+            {
+                VerifyPinConfigured(_interrupt, nameof(InterruptPin));
+                return _gpioController.Read(_interrupt);
+            }
+        }
+
+        /// <summary>
+        /// Reads the current value of Rx0BF pin.
+        /// </summary>
+        public PinValue Rx0BfPin
+        {
+            get
+            {
+                VerifyPinConfigured(_rx0bf, nameof(Rx0BfPin));
+                return _gpioController.Read(_rx0bf);
+            }
+        }
+
+        /// <summary>
+        /// Reads the current value of Rx1BF pin.
+        /// </summary>
+        public PinValue Rx1BfPin
+        {
+            get
+            {
+                VerifyPinConfigured(_rx1bf, nameof(Rx1BfPin));
+                return _gpioController.Read(_rx1bf);
+            }
+        }
+
+        /// <summary>
+        /// Resets the internal registers to the default state and sets Configuration mode.
+        /// </summary>
+        public void Reset()
+        {
+            const byte instructionFormat = 0xC0;
+            _spiDevice.WriteByte(instructionFormat);
+        }
+
+        /// <summary>
+        /// Reads data from the register beginning at the selected address.
+        /// </summary>
+        /// <param name="address">The address to read.</param>
+        /// <returns>The value of address read.</returns>
+        public byte Read(Address address)
+        {
+            const byte instructionFormat = 0x03;
+            const byte dontCare = 0x00;
+            ReadOnlySpan<byte> writeBuffer = stackalloc byte[] { instructionFormat, (byte)address, dontCare };
+            Span<byte> readBuffer = stackalloc byte[3];
+            _spiDevice.TransferFullDuplex(writeBuffer, readBuffer);
+            return readBuffer[2];
+        }
+
+        /// <summary>
+        /// When reading a receive buffer, reduces the overhead of a normal READ
+        /// command by placing the Address Pointer at one of four locations for the receive buffer.
+        /// </summary>
+        /// <param name="addressPointer">The Address Pointer to one of four locations for the receive buffer.</param>
+        /// <param name="byteCount">Number of bytes to read.  This must be one or more to read.</param>
+        /// <returns>The value of address read.</returns>
+        public byte[] ReadRxBuffer(RxBufferAddressPointer addressPointer, int byteCount = 1)
+        {
+            if (byteCount < 1)
+            {
+                throw new ArgumentException($"Invalid number of bytes {byteCount}.", nameof(byteCount));
+            }
+
+            const int StackThreshold = 32;  // Usually won't read more than this at a time.
```

Comment:
might be worth to change to 31 so that you stackalloc multiple of 8

Reply:
will update